notifications-kibana-plugin/public/pages/Channels/Channels.tsx

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  EuiBasicTable,
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiHealth,
+  EuiHorizontalRule,
+  EuiLink,
+  EuiTableFieldDataColumnType,
+  EuiTableSortingType,
+} from '@elastic/eui';
+import { Criteria } from '@elastic/eui/src/components/basic_table/basic_table';
+import { Pagination } from '@elastic/eui/src/components/basic_table/pagination_bar';
+import React, { Component } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { SORT_DIRECTION } from '../../../common';
+import { ChannelItemType, TableState } from '../../../models/interfaces';
+import {
+  ContentPanel,
+  ContentPanelActions,
+} from '../../components/ContentPanel';
+import { CoreServicesContext } from '../../components/coreServices';
+import { BREADCRUMBS, ROUTES } from '../../utils/constants';
+import { DEFAULT_PAGE_SIZE_OPTIONS } from '../Notifications/utils/constants';
+import { ChannelsControls } from './components/ChannelControls';
+import { ChannelsActions } from './components/ChannelsActions';
+
+interface ChannelsProps extends RouteComponentProps {}
+
+interface ChannelsState extends TableState<ChannelItemType> {
+  state: string;
+  type: string;
+  source: string;
+  isActionsPopoverOpen: boolean;
+}
+
+export class Channels extends Component<ChannelsProps, ChannelsState> {
```

Comment:
Any reason we are using `class` here vs `function` in other places?

Reply:
I copied from zhongnan's notification table, which is a class component. But I think in this case class components are easier as there are many table states will update together (if we change `sortField`, `from` should reset to 0). If we use one object to represent table states in a functional component, `useState` won't automatically merge states, seems less readable. And i might be wrong here but if we separate `sortField` and `from` into two `useState`, and when we update them together they might trigger re-render twice

Reply:
Got you- sounds correct on the re-render, I think that can happen. I will have to dive into the notification code base more on my own as well, not familiar with the role of each of these components.